torch/lib/c10d/reducer.cpp

```diff
@@ -429,6 +431,69 @@ void Reducer::set_divide_factor() {
   }
 }
 
+void Reducer::delay_all_reduce() {
+  std::lock_guard<std::mutex> lock(this->mutex_);
+
+  if (!static_graph_ || num_iterations_ > 1) {
```

Comment:
Could we add this check in Python? Advantages are removing the call to C++ and making it clearer to users that it's only called once in static graph training.

Reply:
I feel this is part of reducer internals, it is good to keep it in C++. otherwise we need to expose num_iterations_ to python as well